File: ropchain/gadgets/asm/xor.py
from ropchain.gadgets import util, gadget
from ropchain.gadgets.asm import inc
from ropchain import ropchain

def find(r1, r2, gadgets, canUse):
    xor, canUse = gadget.find(gadgets, canUse, 'xor', r1, r2)
    return util.optROPChain(xor)

File: ropchain/lib/src/asm/xor.cpp
#include "xor.h"

OptROP Xor::find(const Opcode& op1, const Opcode& op2,
        const Gadgets& gadgets, RegSet& aval) {
    auto gadget = Util::find(gadgets, aval, "xor", op1, op2);
    return Util::toOptROP(gadget);
}
